### DISCUSSION TITLE
FX's TODO List

### DISCUSSION THREAD

**Discussion by FxMorin** [FxMorin/MoreCulling#67] [Ideas]
**Things to cull:**
- Entity Culling
- Inventory item models should cull invisible side
- Shulker entities should cull like a full blocks 👀 

**Discarded Idea's:**
- Block Entity Culling (Not worth it)

**Comment by PweSol** [comment]
> Shulker entities should act like full blocks

Sound like something [EBE](https://github.com/FoundationGames/EnhancedBlockEntities) is doing. Hopefully, these two mods can stay compatible with each other. If necessary, EBE allows shulker optimizations to be disabled trough the config.

**Comment by ghost** [comment]
Isn't Entity Culling the same thing the Entity Culling mod does? Or do you mean against block faces?

**Reply by FxMorin** [reply]
I just meant entity culling in general. So both, cause I want to attempt to make my own culling techniques. 
An example would be, culling the bottom of the entities feet if they are on a solid block. Since that block is already cached by the entity so it should be really fast and remove 2+ faces for all entities on ground

**Reply by ghost** [reply]
Woah, that'd be cool!

I think adding this sort of culling may be difficut because of entity optimizations, though. 

Could there be 3-face culling for entity voxels?

If you are also going to implement a (better) alternative to the existing EntityCulling mod, will there be enough customization? With the existing mod you can customize the delay between culling passes, if nametags are shown through walls, specify an entity whitelist for content mods and more.

**Reply by FxMorin** [reply]
> I think adding this sort of culling may be difficut because of entity optimizations, though.

Shouldn't change anything for me, you can be more specific if you want to hear me reasoning why tho.

> Could there be 3-face culling for entity voxels?

The 3-face culling works due to the item frames being axis-aligned. Entities are not. Although I've been working on some changes that may allow it to work on non axis-aligned things. So maybe I will be able to implement it there.
It all comes down to: `Do I get a performance win: yes or no`

> If you are also going to implement a (better) alternative to the existing EntityCulling mod, will there be enough customization?

That mod does things Async so you need the customization.
Usually with my changes you shouldn't see the change (cause of you do it means its not working correctly)
You will always be able to toggle off features tho, it always depends how I implement it.
